### ISSUE TITLE
could not connect to server

### ISSUE THREAD

**Issue by UriZafrir** [TuringSoftware/CrystalFetch#27]
Hi
When using with Mac M1 and pressing the "all builds" button I get "could not connect to server

![image](https://github.com/TuringSoftware/CrystalFetch/assets/35652384/6aafadcf-255b-4d40-aedf-dd307efceeaf)
![image](https://github.com/TuringSoftware/CrystalFetch/assets/35652384/3f4ff90a-a787-497f-b188-5691e9d59f78)

**Comment by chadlwilson** [comment]
Not sure if its the same issue, but I always just get timeouts trying to load the "All builds..." list with `2.1.0 (4)`.

![image](https://github.com/TuringSoftware/CrystalFetch/assets/29788154/d3680a88-c0c4-4509-9cbf-91b75687c3aa)

**Comment by chadlwilson** [comment]
Ahh, looks like UUPDump has been down for the last 2 weeks or so: https://www.reddit.com/r/windowsinsiders/comments/16hci7n/is_uupdump_down_for_everyone/

There is https://www.uup.ee/ so I wonder if there is a way to point CrystalFetch at a different UUPDump site and/or a local instance.

**Comment by Kusanalee** [comment]
UUPDump is still down. There should be an option to use alternative uup sites.

**Comment by Kusanalee** [comment]
So I have tried to switch it out with UUPDump.ee - but that doesn't work because it doesn't have a json-api. I also tried uupdump.cn which does have a json-api but crystalfetch still spits out a "the data couldn't be read because" it isn't in the correct forma". I'm currently looking for other sources of UUP in hopes that it works.
<img width="418" alt="image" src="https://github.com/TuringSoftware/CrystalFetch/assets/84009428/a7f2130a-73ef-4781-a650-9ddc17849e47">

**Comment by Helloyunho** [comment]
![image](https://github.com/TuringSoftware/CrystalFetch/assets/10192625/19f08149-6f7e-4965-9732-740324d27381)
https://uupdump.net seems like permanently down now...

**Comment by Helloyunho** [comment]
> ![image](https://user-images.githubusercontent.com/10192625/273116105-19f08149-6f7e-4965-9732-740324d27381.png) https://uupdump.net seems like permanently down now...

You can download the whole website source from there and use the JSON API via the local network though hmmm...
I think we need an option to select our own API endpoint URL

**Comment by chadlwilson** [comment]
After some hacking around with both the UUPDump JSON API and CrystalFetch (to change it to point to a local version of the API running in a Docker instance) I did get a setup running locally that allowed me to download an experimental build.

I'll see if I can automate the UUPDump "distribution" stuff to make this work, and see if I can submit a PR to CrystalFetch to allow the UUPDump URL to be changed.

Broad steps (which needs more detail)
* Get UUPDump into the right state (PHP `8.2` won't work OOTB due to some deprecations/warnings)
  ```
  git clone git@github.com:uup-dump/json-api.git --depth=1
  cd json-api
  git submodule init && git submodule update
  git clone git@github.com:uup-dump/fileinfo.git --depth=1
  git clone git@github.com:uup-dump/packs.git --depth=1
  docker run -d -p 80:80 --name uupdump-json-api -v "$PWD":/var/www/html php:8.1-apache
  ```
* Download and build CrystalFetch per the README
* Change URL to `http://127.0.0.1`
https://github.com/TuringSoftware/CrystalFetch/blob/d0fbf0fe668c2966069d99e65fefc296d73b4bfd/Source/UUPDump/UUPDumpAPI.swift#L23
* Now it seems to all work and allow me to build ISOs.

**Comment by gedw99** [comment]
Is this the same thing ?

https://www.uupdump.cn/?lang=en-us

Seems to have various build of Windows OS, etc

**Comment by Kusanalee** [comment]
I tried using uupdump.cn, however that does not work (usually with an error about the network connection was lost). I'll try locally hosting it in docker to see if works.

**Comment by chadlwilson** [comment]
Yeah, it definitely works, although the data is slightly stale in uupdump's repos, so it's missing a few builds. Looks like the auto-updates of data into GitHub is also broken. e.g at time of writing it is missing [this build](https://blogs.windows.com/windows-insider/2023/10/05/announcing-windows-11-insider-preview-build-25967-canary-channel/))

This may reduce the usefulness of even running locally over time unless there are enough folks doing manual PRs.
https://github.com/uup-dump/packs/commits/master
https://github.com/uup-dump/fileinfo/commits/master

**Comment by chadlwilson** [comment]
My PR to allow this to be configured is at https://github.com/TuringSoftware/CrystalFetch/pull/32 - not much help since you'd still need to build locally unless merged/released and could as easily edit the URL in the code for your local use, but might be a starting point for someone who knows SwiftUI better than I 🙏

**Comment by Helloyunho** [comment]
![CleanShot 2023-10-30 at 08 57 07@2x](https://github.com/TuringSoftware/CrystalFetch/assets/10192625/f7a2bdb9-5831-4453-a43e-ccd7fc83da53)
Great news! About a week ago https://uupdump.net is open again!
...But as mentioned in #32 uupdump creates wrong iso for arm64
